src/Tracker/index.tsx

```diff
@@ -126,24 +126,14 @@ const trackComponent = <P extends WithTrackingProps>(
   class TrackedComponent extends Component<P> {
     callback?: () => AnalyticsDynamicProperties = undefined
     componentDidMount() {
-      const properties =
-        typeof this.callback === 'function' ? this.callback() : undefined
+      const properties = this.props.trackPropertiesBeforeMount
```

Comment:
Would add `typeof this.props.trackPropertiesBeforeMount === 'function'` to be safe

Reply:
Do you think it can be changed at runtime ? Because at compile time, it will be enforced with the type WithTrackingProps:
```
export type TrackPropertiesBeforeMountCallback = () => Record<string, unknown>

export type WithTrackingProps = {
  trackScreen: TrackScreenCallback
  trackPropertiesBeforeMount?: TrackPropertiesBeforeMountCallback
}
```